src/widgets/buttons/DrawnButton.cpp

```diff
@@ -0,0 +1,108 @@
+#include "widgets/buttons/DrawnButton.hpp"
+
+#include "singletons/Theme.hpp"
+
+#include <QPainter>
+
+namespace chatterino {
+
+DrawnButton::DrawnButton(Type type_, Options options, BaseWidget *parent)
+    : Button(parent)
+    , foreground(this->theme->messages.textColors.system)
+    , foregroundHover(this->theme->messages.textColors.regular)
+    , basePadding(options.padding)
+    , baseThickness(options.thickness)
+    , type(type_)
+{
+    this->setContentCacheEnabled(true);
+}
+
+void DrawnButton::setBackground(QColor color)
+{
+    this->background = color;
+    this->invalidateContent();
+}
+
+void DrawnButton::setBackgroundHover(QColor color)
+{
+    this->backgroundHover = color;
+    this->invalidateContent();
+}
+
+void DrawnButton::setForeground(QColor color)
+{
+    this->foreground = color;
+    this->invalidateContent();
+}
+
+void DrawnButton::setForegroundHover(QColor color)
+{
+    this->foregroundHover = color;
+    this->invalidateContent();
+}
+
+void DrawnButton::mouseOverUpdated()
+{
+    this->invalidateContent();
+}
+
+void DrawnButton::paintContent(QPainter &painter)
+{
+    QColor fg;
+    QColor bg;
+
+    if (this->mouseOver())
+    {
+        bg = this->backgroundHover;
+        fg = this->foregroundHover;
+    }
+    else
+    {
+        bg = this->background;
+        fg = this->foreground;
+    }
+
+    if (bg.isValid())
+    {
+        painter.fillRect(this->rect(), bg);
+    }
+
+    switch (this->type)
+    {
+        case Type::Plus: {
+            QPen pen;
+            pen.setColor(fg);
+            int thickness = std::max(
+                1,
+                static_cast<int>(std::round(
+                    static_cast<double>(this->baseThickness) * this->scale())));
+            pen.setWidth(thickness);
+            painter.setPen(pen);
+
+            auto innerSize = this->rect().size();
+            innerSize.setHeight(innerSize.width());
+            QRect inner;
+            inner.setSize(innerSize);
+            inner.moveCenter(this->rect().center());
+
+            int padding = static_cast<int>(
+                static_cast<float>(this->basePadding) * this->scale());
```

Comment:
The thickness is rounded, but the padding isn't. Is that intended? Also, for future types, it might be helpful to move these to a separate method.

Reply:
Padding and line thickness were moved to private methods in [`bb22455` (#6349)](https://github.com/Chatterino/chatterino2/pull/6349/commits/bb224555976d22d7be4e27fc330440da1638bc18)

I made padding round in [`8266638` (#6349)](https://github.com/Chatterino/chatterino2/pull/6349/commits/8266638363143dd318d92d3d017c8a234682f828)